src/patternfly/components/Table/table.scss

```diff
@@ -1054,6 +1049,15 @@
 .#{$table}__tbody {
   vertical-align: top;
 
+  .#{$table}__sort {
+    vertical-align: top;
+  }
```

Comment:
If the headers vary in height, this is going to make the sortable ones top aligned

<img width="1251" alt="Screenshot 2024-12-03 at 2 41 59 PM" src="https://github.com/user-attachments/assets/3bd85826-132c-4856-a10c-997c0c760f08">